syntax = "proto3";

enum ShipCategory {
    Un = 0;
    Ally = 1;
    Enemy = 2;
}

enum ShipNames {
    Un1 = 0;
    Corvette = 1;
    Frigate = 2;
    Cruiser = 3;
    Destroyer = 4;
    Aircraft_Carrier = 5;
    Dreadnought = 6;
}

message Officers {
    string first_name = 1;
    string last_name = 2;
    string rank = 3;
}

message ForRequest {
    int32 number = 1;
}

message ShipRequest {
    ShipCategory alignment = 1;
    string name = 2;
    ShipNames classes = 3;
    double length = 4;
    int32 crew_size = 5;
    bool armed = 6;
    repeated Officers officers = 7;
}

service Recommendations {
    rpc GetShip(ForRequest) returns (ShipRequest);
}